Gives Psychologist a Transorbital Lobotomy Kit.

XRandomXManX opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

As well as putting all of the Psychologist's Medications into a box, this PR gives them a fully functional Transorbital Lobotomy Kit!
Adds a Orbitoclast, Hammer, and a Transorbital Lobotomy procedure.

## Why It's Good For The Game

This kit will allow Psychologists to both cure and give brain traumas.
For anyone willing to be patient, after enough Lobotomies they might be able to get a superpower like being able to walk in crit, or have RNG punching power, or, if they're lucky, an Imaginary Friend.
I don't think this will cause balance issues since the Psychologist cannot Lobotomize themselves, and not many people will wait for the Psychologist to give them 6 back to back lobotomies.

This could be a good way of giving Psychologists who feel up to the task something useful to do if nobody wants to talk about their feelings.

## Changelog
:cl: Radacitus
tweak: put Medications into Medication Box
imageadd: Added Orbitoclast and Hammer sprites
add: Added Orbitoclast and Hammer tools
imageadd: Added Case sprites
add: Adds Lobotomy Kit
add: Adds Transorbital Lobotomy procedure
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
moves all of the psychologists prescriptions to a box

Commit:
adds hammer and orbitoclast sprites

Commit:
adds in orbitoclast and hammer tools

Commit:
gives the hammer a significant buff

Commit:
adds in sprite for lobotomy tools case

Commit:
adds lobotomy kit, adds weight class for hammer

Commit:
fixes many small misprints, adds briefcase to psycologists bag

Commit:
adds in transorbital lobotomy procedure. Last touches as well.

uomo91 commented on code/modules/surgery/organic_steps.dm:
```diff
@@ -138,3 +138,37 @@
+	display_results(user, target, "<span class='notice'>You hammer past [target]'s [parse_zone(target_zone)].</span>",
+		"<span class='notice'>[user] hammers past [target]'s [parse_zone(target_zone)]!</span>",
+		"<span class='notice'>[user] hammers past [target]'s [parse_zone(target_zone)]!</span>")
+	return ..()
```
```suggestion
	return ..()

```

Add a new line to the file ending here.

uomo91 commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -0,0 +1,67 @@
+				target.gain_trauma_type(BRAIN_TRAUMA_SPECIAL, TRAUMA_RESILIENCE_MAGIC)
+	else
+		user.visible_message("<span class='warning'>[user] suddenly notices that the brain [user.p_they()] [user.p_were()] working on is not there anymore.</span>", "<span class='warning'>You suddenly notice that the brain you were working on is not there anymore.</span>")
+	return FALSE
```
```suggestion
	return FALSE

```

Add a new line to the file ending here.

uomo91 commented on code/game/objects/items/storage/briefcase.dm:
```diff
@@ -48,3 +48,24 @@
+
+/obj/item/paper/guides/lobotomy
+	name = "Lobotomies For Dummies"
+	info = "Doing a Lobotomy is easy.<br> Target the eyes, use your drapes, orbitoclast, hammer, orbitoclast, then finish off with a cautery.<br> However, to retrieve special powers from this is a little harder.<br> You must fail the lobotomizing step (second orbitoclast) to cause some traumas.<br> If these traumas are too bad, do the step correctly then fail it again.<br> To fail a step on purpose, ...
```
```suggestion
	info = "Doing a Lobotomy is easy.<br> Target the eyes, use your drapes, orbitoclast, hammer, orbitoclast, then finish off with a cautery.<br> However, to retrieve special powers from this is a little harder.<br> You must fail the lobotomizing step (second orbitoclast) to cause some traumas.<br> If these traumas are too bad, do the step correctly then fail it again.<br> To fail a step on purpose, you must be on disarm intent."

```

Add a new line to the file ending here.

uomo91 reviewed (changes requested):
Looks neat!

Commit:
adds an empty line at the end of organic steps

Co-authored-by: uomo <51800976+uomo91@users.noreply.github.com>

Commit:
adds an empty line at the end of briefcase.dm

Co-authored-by: uomo <51800976+uomo91@users.noreply.github.com>

Commit:
adds a line at the end of transorbital_lobotomy.dm

Co-authored-by: uomo <51800976+uomo91@users.noreply.github.com>

wesoda25 commented:
“Purposefully fucking up eye surgery” would like to have a word with you

ATH1909 commented on code/modules/surgery/tools.dm:
```diff
@@ -442,3 +442,37 @@
+	attack_verb_continuous = list("tries to lobotomize", "stabs")
+	attack_verb_simple = list("try to lobotomize", "stab")
+	tool_behaviour = TOOL_ORBITOCLAST
+	toolspeed = 1
```
make this able to inflict puncturing wounds

ATH1909 commented:
> “Purposefully fucking up eye surgery” would like to have a word with you

This.

Tbh, I don't see why we should add two new surgical tools that are only used in a single kind of surgery that's somewhat redundant with a failure result of a surgery we already have, especially since you haven't added any ways to prints or buy these tools (you apparently can only get them by entering the round as a psychologist).

A better PR would be to cause failing eye surgery to (have a chance to) directly add traumas (like the lobotomy surgery's failure result) in addition to dealing the 80 brain damage that it already does.

XDTM commented:
Basically this is a replica of the lobotomy surgery, but uses different tools and is faster, right?

One thing i would change is making the fluff text for success and failure the same, and a bit more generic, like "You complete the transorbital lobotomy procedure.", to give it a downside. Honestly i'd increase the chance of bad outcomes a bit so it doesn't make regular lobotomy completely obsolete.

Winterous commented:
You mention a "ignore damage slowdown" trauma, but this doesn't add it; were you under the impression that one already exists? Am I missing it somehow?
There's one that makes you immune to crit, one that makes you feel no pain and regen stamina faster.

Anyway, the instructions imply that you can cure the traumas you inflict by messing the step up; that's not true, they're all cured by magic only.

XRandomXManX commented:
First of all, the items are meant to be Psychologist only, the same way the Chaplain uses the Null Rod or the Curator can buy their costumes. This also stops it from making normal lobotomies obsolete, since only the person with these tools will be able to do them Transorbitally. Yes, I was talking about the already existing Traumas, I did not add any yet. No, I didn't say you could get rid of traumas by failing the surgery, that's how you get them. You get rid of traumas by doing the surgery correctly. I don't understand how this would fuck up Eye Surgery. Eye Surgery is fine, as far as I know. This is a different type of surgery.

Winterous commented:
Sorry I worded that poorly.
The instructions imply that by doing the surgery correctly, you can cure the traumas you inflict by messing the step up. That's not true, as they have trauma_resilience_magic.

XRandomXManX commented:
OH! I'm not sure if I want to buff this any further, unless y'all think that's okay. It seems like this would place a great block on the usefulness of how I wanted the Psychologist to be played. I honestly don't think I could get away with buffing this any amount with the way it's being received. I at least know that SOMEONE will come in and take the risk of getting some effed up brain traumas. Anyways, I'm gonna go fix some more things about this PR now and be sad.

Commit:
removes repeat lines of code

XRandomXManX commented:
Nevermind, just tested it and it's fine. Honestly considering debuffing it now. I'm gonna make it so mild traumas can't be cured, ones like restricting the players speech to the 1000 most common words or Nervous Coughing or Dumbness. By can't be cured I mean can't be cured by Lobotomy, they could probably still be cured by some chems.

Commit:
Mild Traumas caused by failed Transorbital Lobotomies can no longer be cured by more Lobotomies

Commit:
Fixes a misprint

Fixes a misprint that caused the initial hammering step to be described as picking to the player

Commit:
Orbitoclast is now pointy

Commit:
Merge branch 'psych-upgrade' of https://github.com/XRandomXManX/tgstation-1 into psych-upgrade

XRandomXManX commented:
Not sure what 'Merge branch 'psych-upgrade' of' is about, but it just seems to be readding changes I already made so I think it's fine. Other than that everything looks good now. I hope y'all can see that these aren't meant to be new surgery tools, just a tool for the Psychologist. Although the surgery is universal so anyone can try ghetto lobotomies, even though they'll almost never succeed them unless they have an e-sword, making them much more punishing.

ATH1909 commented:
> I don't understand how this would fuck up Eye Surgery. Eye Surgery is fine, as far as I know. This is a different type of surgery.

I never said it would "fuck up" eye surgery, I just said that intentionally failing eye surgery using disarm intent does basically what this surgery is trying to do (intentionally damage someone's brain by going at it through their eyes), making it redundant. I feel like just modifying the failure result of eye surgery to directly give traumas in addition to the brain damage it deals would be an easier thing to do than adding two new surgical tools and an entire new surgery for this.

XRandomXManX commented:
No. I'm not shoving everything into Eye Surgery because you don't want new items in the game. There's nothing wrong with adding new tools; it's meant to be the Psychologists entire gimmick right now. The new surgery is there so people can use ghetto tools to do failed lobotomies. If I JUST wanted to add a new way to get traumas I'd do the Eye Surgery thing, but I want to give them something that an assistant couldn't do, that doesn't do 80 brain damage.

XDTM commented:
@ATH1909 this is literally the lobotomy surgery with different tools and steps, there's no point in making a comparison with eye surgery when you can compare it to lobotomy itself

@XRandomXManX the resistance tier indicates the level of cure necessary to cure the trauma, so TRAUMA_RESILIENCE_MAGIC means that only magic or above can cure it. Your last change lowered it making it possible for lobotomies to cure mild traumas, which i suppose is the opposite of what you intended.

For reference the scale is
TRAUMA_RESILIENCE_BASIC (Neurine)
TRAUMA_RESILIENCE_SURGERY (Brain Surgery)
TRAUMA_RESILIENCE_LOBOTOMY (Lobotomy)
TRAUMA_RESILIENCE_MAGIC (Staff of healing/Adminheals)
TRAUMA_RESILIENCE_ABSOLUTE (only used for special cases where it must not be cured except by one specific method)

XRandomXManX commented:
But I did it, I cured Traumas with Lobotomy from Lobotomy. I don't understaaaaaaaaaaaaaaaaaaaaaaaaaaaand

XRandomXManX commented:
gonna keep testing, I guess. Maybe somebody broke the system at some point and nobody noticed

Commit:
switches the cure types with mild and severe

because I guess I was in some sort of freak accident the first time

XRandomXManX commented:
Everything is good and working, you can merge it now. Do it, please, before you change your minds.

Rohesie commented on code/modules/surgery/organic_steps.dm:
```diff
@@ -138,3 +138,37 @@
+/datum/surgery_step/pick
+	name = "pick into bone"
+	implements = list(TOOL_ORBITOCLAST = 100, TOOL_SCREWDRIVER = 25, /obj/item/stack/rods = 10)
+	time = 30
```
```suggestion
	time = 3 SECONDS
```

Rohesie commented on code/modules/surgery/organic_steps.dm:
```diff
@@ -138,3 +138,37 @@
+/datum/surgery_step/drive
+	name = "drive through bone"
+	implements = list(TOOL_HAMMER = 100, /obj/item = 20)
+	time = 30
```
```suggestion
	time = 3 SECONDS
```

Rohesie commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -0,0 +1,64 @@
+	target_mobtypes = list(/mob/living/carbon/human, /mob/living/carbon/monkey)
+	possible_locs = list(BODY_ZONE_PRECISE_EYES)
+
+/datum/surgery/transorbital_lobotomy/can_start(mob/user, mob/living/carbon/target)
+	if(!..())
+		return FALSE
```
```suggestion
/datum/surgery/transorbital_lobotomy/can_start(mob/user, mob/living/carbon/target)
	. = ..()
	if(!.)
		return
```

Rohesie commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -0,0 +1,64 @@
+/datum/surgery/transorbital_lobotomy/can_start(mob/user, mob/living/carbon/target)
+	if(!..())
+		return FALSE
+	var/obj/item/organ/brain/B = target.getorganslot(ORGAN_SLOT_BRAIN)
+	if(!B)
+		return FALSE
+	return TRUE
```
```suggestion
	return target.getorganslot(ORGAN_SLOT_BRAIN) ? TRUE : FALSE

```

Rohesie commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -0,0 +1,64 @@
+			"<span class='notice'>[user] completes the surgery on [target]'s brain.</span>")
+
+	target.cure_all_traumas(TRAUMA_RESILIENCE_LOBOTOMY)
+	if(target.mind && target.mind.has_antag_datum(/datum/antagonist/brainwashed))
```
```suggestion
	if(target.mind?.has_antag_datum(/datum/antagonist/brainwashed))
```

Rohesie commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -0,0 +1,64 @@
+/datum/surgery_step/tlobotomize
+	name = "perform transorbital lobotomy"
+	implements = list(TOOL_ORBITOCLAST = 75, /obj/item/melee/transforming/energy/sword = 50, /obj/item/stack/rods = 10)
+	time = 75
```
```suggestion
	time = 7.5 SECONDS
```

Rohesie commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -0,0 +1,64 @@
+	return ..()
+
+/datum/surgery_step/tlobotomize/failure(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/obj/item/organ/brain/B = target.getorganslot(ORGAN_SLOT_BRAIN)
+	if(B)
```
Do an early return here, please, with `if(!B)` instead.
And if you could use a better variable name than `B`, that'd be better for clarity.

XRandomXManX replied:
I just copy pasted and buffed the lobotomy procedure, but I'll try to work this out if I can.

XRandomXManX replied:
the deed is done

Rohesie commented:
Did you add an empty icon to `icons/obj/surgery.dmi` ?

Winterous commented:
> 
> 
> But I did it, I cured Traumas with Lobotomy from Lobotomy. I don't understaaaaaaaaaaaaaaaaaaaaaaaaaaaand

`		if(1)
			target.gain_trauma_type(BRAIN_TRAUMA_MILD, TRAUMA_RESILIENCE_MAGIC)
		if(2)
			if(HAS_TRAIT(target, TRAIT_SPECIAL_TRAUMA_BOOST) && prob(50))
				target.gain_trauma_type(BRAIN_TRAUMA_SPECIAL, TRAUMA_RESILIENCE_MAGIC)
			else
				target.gain_trauma_type(BRAIN_TRAUMA_SEVERE, TRAUMA_RESILIENCE_MAGIC)
		if(3)
			target.gain_trauma_type(BRAIN_TRAUMA_SPECIAL, TRAUMA_RESILIENCE_MAGIC)`

This is what the normal lobotomy does, whether you pass or fail the step; all of them are trauma_resilience_magic, meaning they can't be cured with lobotomy.

` 	info = "Doing a Lobotomy is easy.<br> Target the eyes, use your drapes, orbitoclast, hammer, orbitoclast, then finish off with a cautery.<br> However, to retrieve special powers from this is a little harder.<br> You must fail the lobotomizing step (second orbitoclast) to cause some traumas.<br> **If these traumas are too bad, do the step correctly then fail it again.**<br> To fail a step on purpose, you must be on disarm intent."`

The bold part suggests that you can FIX the 'mistake' traumas by doing the step correctly. That's not supposed to be possible.

Lobotomy REMOVES a chunk of the brain, causing irreparable damage (and thus incurable traumas), in order to remove severely damaged parts of the brain that are causing otherwise-incurable traumas.

https://github.com/tgstation/tgstation/pull/54418/files#diff-d1399521219a02e667d84998513cbd057f261ccbef2e16d8cdf59807f323edcfR40

This line, by having trauma_resilience_lobotomy instead of trauma_resilience_magic, is making those severe traumas inflicted by it curable via lobotomy.

.

To clarify: The instructions should not suggest that you can cure the damage you cause with the procedure. And, you should not be ABLE to cure the damage you cause with the procedure.

Tlaltecuhtli commented:
does psych have surgery room access?

Commit:
timechange1

Co-authored-by: Rohesie <rohesie@gmail.com>

Commit:
Update code/modules/surgery/transorbital_lobotomy.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

Commit:
Update code/modules/surgery/organic_steps.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

Commit:
Update code/modules/surgery/transorbital_lobotomy.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

Commit:
Update code/modules/surgery/transorbital_lobotomy.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

Commit:
Update code/modules/surgery/transorbital_lobotomy.dm

Co-authored-by: Rohesie <rohesie@gmail.com>

XRandomXManX commented:
> does psych have surgery room access?

No, but they'll start with a surgery kit now.

XRandomXManX commented:
The reason why the Psychologist can cure severe traumas caused by lobotomies is so they don't give people permanent blindness by accident, not because it's realistic. The same way a lobotomy would not cure any sort of brain disorder at all in real life.

XRandomXManX commented:
> Did you add an empty icon to `icons/obj/surgery.dmi` ?

No.

Commit:
fix

Commit:
spaces intead of tabs moment

Commit:
brian moment

XRandomXManX commented:
Also, adding onto the curable severe traumas thing. Traumas don't directly correlate to parts of the brain removed. Just because you gain Monophobia, and die of fear when not in the presence of others doesn't mean that couldn't be cured by just removing even more of the brain. Of course none of this could be done to any degree of accuracy without completely destroying a person, which is way it's seen as such a horrible procedure and why it isn't in any way realistic.

Bawhoppen commented:
It doesn't feel like this really fits in the psychologist's arsenal... the psychologist is more like a therapist, not really a neuro-surgeon. I think this might be slightly misplaced of an addition.

Winterous commented:
But now you're just making this procedure SUPERIOR to the regular lobotomy, because regular one is ALL trauma_resilience_magic.

The point is that by removing a chunk of the brain, there are consequences of incurable brain damage (incurable because part of the brain is missing and cannot be replaced or regrown). Why would removing ANOTHER chunk of the brain heal that?
In order to repair that trauma caused by removing chunks, you'd need to REGROW the chunk, which can only be achieved by magical effects. That's the point.

XDTM commented:
Leaving aside realism for a moment, lobotomy is meant to be a "last resort" procedure for traumas that are beyond regular surgery. The idea is that it requires people to evaluate if they'd rather live with their trauma, or if they want to risk potentially getting an even worse one _permanently_ to cure it. Making any of it reversible undermines this decision, so i'd avoid that approach.

XRandomXManX commented:
This is the Psychologists only ability, this isn't overpowered. They can't become Hulks or have Xenobio God Powers. Only the Psychologist gets this surgery set, so it doesn't matter that it's better than normal surgery. This doesn't undermine the Psychologist's Role as it is now, because the Psychologist became a Psychiatrist once they were given drugs to prescribe. They don't have any set role, that's up to the person playing them. I said that since the lobotomy in the game already works completely different than real life, it wouldn't be out of the ordinary for something cartoonish like severing the connections between even more parts of the brain to fix the trauma. As long as you're using the Psychologist's advanced tools, that is.

XRandomXManX commented:
Also you don't have to be a neurosurgeon to do a transorbital lobotomy, you just need to know what angle to hold the ice pick at to slide it between the frontal lobe and the rest of the brain. It's not a very good surgery. Which is why in real life it isn't a last resort, because it's worse than normal brain surgery or just doing nothing at all.

Rohesie commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -42,22 +42,22 @@
-		user.visible_message("<span class='warning'>[user] suddenly notices that the brain [user.p_they()] [user.p_were()] working on is not there anymore.</span>", "<span class='warning'>You suddenly notice that the brain you were working on is not there anymore.</span>")
-	return FALSE
+    var/obj/item/organ/brain/Brian = target.getorganslot(ORGAN_SLOT_BRAIN)
+    if(!Brain)
```
This is `Brian` first and `Brain` after. Subtle difference.
Use snake_case for variables if you can, instead of PascalCase

Winterous commented:
So if it's such a crude procedure, why are you trying to defend it being mechanically superior to its surgical counterpart when it should be a purely flavourful thing in the first place?

Rohesie commented on code/modules/surgery/organic_steps.dm:
```diff
@@ -138,3 +138,37 @@
 		"<span class='notice'>[user] drills into [target]'s [parse_zone(target_zone)]!</span>")
 	return ..()
+
+//pick bone
```
```suggestion

```
Not very informative.

XRandomXManX replied:
all of the surgery steps are commented on like this so they can be seen better when you're designing a new surgery and need to find which steps it should have.

Rohesie commented on code/modules/surgery/organic_steps.dm:
```diff
@@ -138,3 +138,37 @@
+		"<span class='notice'>[user] picks into [target]'s [parse_zone(target_zone)]!</span>")
+	return ..()
+
+//drive through bone
```
```suggestion

```
Not very informative.

Rohesie commented on code/modules/surgery/organic_steps.dm:
```diff
@@ -138,3 +138,37 @@
+//drive through bone
+/datum/surgery_step/drive
+	name = "drive through bone"
+	implements = list(TOOL_HAMMER = 100, /obj/item = 20)
```
Is this supposed to be any sharp item? That's usually implemented with a `tool_check`

XRandomXManX replied:
Thank you. I am trying to fix this now.

Rohesie commented on code/modules/surgery/transorbital_lobotomy.dm:
```diff
@@ -0,0 +1,63 @@
+	return ..()
+
+/datum/surgery_step/tlobotomize/failure(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/obj/item/organ/brain/Brian = target.getorganslot(ORGAN_SLOT_BRAIN) //Brian is the patient's brain, Brain, Brian.
+	if(!Brian)
```
use snake_case instead of PascalCase for variables
btw, funny as it may be, Brian is not really very clear
use something like `brain_organ`

XRandomXManX replied:
I'm renaming it to target_brain now, because it is the target's brain. I thought I could get away with a funny variable if I commented on it's function :P

XRandomXManX commented:
> So if it's such a crude procedure, why are you trying to defend it being mechanically superior to its surgical counterpart when it should be a purely flavourful thing in the first place?

Stop arguing. It's just something for the Psychologist to do. Stop trying to argue that it shouldn't be added into the game because it's unrealistic. Nothing about any of the surgeries in the game are realistic. Try to argue about something that actually affects the game instead of whether or not it's realistic or superior to the normal lobotomy. It's meant to be better than the normal lobotomy. The normal lobotomy wouldn't be any different in real life, but it is so the Psychologist can have a unique function.

Winterous commented:
Except that the way in which it's better that we're objecting to is ridiculous and bizarrely inconsistent; I don't care that it's faster to do that the regular one, just that for some strange reason ONE of the cases in which a trauma is inflicted can be cured by another lobotomy.
I would honestly care less if ALL of them were lobotomy-curable.

XRandomXManX commented:
I'm going to make both Lobotomies have curable Severe Traumas

Commit:
fixes inconsistent things, tool needs >4 force for drive

Winterous commented:
Why ONLY the severe traumas though? Why not the mild and special ones too? It makes no sense.

XRandomXManX commented:
Because I want the players to have mild traumas. Maybe they are too small to pin down? It's at least consistent in-game now.

XRandomXManX commented:
Also if I removed special traumas it would require way more re-rolling to get a good combo

XRandomXManX commented:
I'll remove mild traumas, now that I'm looking at how some of the special traumas are not so good. Those ones just can't be removed because they're special.

Commit:
Update transorbital_lobotomy.dm

Commit:
Update lobotomy.dm

XRandomXManX commented:
Next PR I'll add more special traumas, some good, some bad. So you don't have to worry about balance too much.

Winterous commented:
People shouldn't be using lobotomy to repeatedly roll for interesting traumas in the first place, because the entire point is that it's permanent damage!

XRandomXManX commented:
Some Special Traumas are damaging too. Either way I have disliked every addition you've screamed at me to make, they've all made the PR worse than it was when it started. I don't care that you want someone to be incurably blind for an entire round because "That's how it should be". This is stupid. Stop asking me to make changes with no reason other than wanting it to be that way, wanting it to be 'realistic', or your newest argument being 'the point'. I don't care about the point, it's a new point. Stop trying to nerf the weakest job in the entire game. I do not care what it was before. This is adding something new. Just stop getting so invested in my PR, asking me to make changes when all you want is for the PR not to exist. If you have no actual feedback stop asking me to make changes.

Commit:
removes lobotomy

Commit:
Merge branch 'psych-upgrade' of https://github.com/XRandomXManX/tgstation-1 into psych-upgrade

XRandomXManX commented:
Now only the Psychologist can do lobotomies. Crew can still do lobotomies, the probability is just lower since they'd have to use ghetto tools.

Commit:
Delete transorbital_lobotomy.dm

XRandomXManX commented:
github, stop. Please

XRandomXManX commented:
I am going to bash my head into a table

Commit:
Revert "Merge branch 'psych-upgrade' of https://github.com/XRandomXManX/tgstation-1 into psych-upgrade"

This reverts commit 68ae95817fd85df9606940fd228928cca258396e, reversing
changes made to 9487f6fc8467e92e807aef6a318d1bad1e992f6d.

Commit:
Merge branch 'psych-upgrade' of https://github.com/XRandomXManX/tgstation-1 into psych-upgrade

Commit:
Update transorbital_lobotomy.dm

XRandomXManX commented:
I just have to remove the references to the lobotomy now

Commit:
removes references to lobotomy

Commit:
Merge branch 'psych-upgrade' of https://github.com/XRandomXManX/tgstation-1 into psych-upgrade

XRandomXManX commented:
That's it, that's the change. Lobotomies now have reversible effects and only the Psychologist can do them. This is the PR.

Commit:
i forgot to remove a reference

Rohesie commented on code/modules/surgery/organic_steps.dm:
```diff
@@ -138,3 +138,42 @@
+		"<span class='notice'>[user] begins to hammer through the bone past [target]'s [parse_zone(target_zone)].</span>",
+		"<span class='notice'>[user] begins to hammer through the bone past [target]'s [parse_zone(target_zone)].</span>")
+
+/datum/surgery_step/drive/tool_check(mob/user, obj/item/tool)
+	if(implement_type == /obj/item && tool.force > 4)
+		return FALSE
+	return TRUE
```
This proc doesn't do what you want it to. We use `istype()` checks for type comparisons, but in this case what you'd want to check is whether the tool is not a hammer.
Also, this is returning false if the tool has force.

Rohesie commented:
Could you make the tools orderable through cargo, and to move part of the operation's success chance into a skillchip for the psychologist?

XRandomXManX commented:
> Could you make the tools orderable through cargo, and to move part of the operation's success chance into a skillchip for the psychologist?

Sure.

Commit:
No item, only drill

It is only 25% because you're hammering with it, dummy

Commit:
you can now order lobotomy tools from cargo

You can't order the case because the case has the ability to KO the same as a briefcase, making it a fairly good weapon, even if it does have reduced damage.

XRandomXManX commented:
Now to attempt skillchip

Commit:
adds groundwork for lobotomizer skillchip

I just don't know how to make it work! I want it to cause any lobotomy to work with any object. Just the final step.

XDTM commented:
I made special traumas with the explicit intention of making them a rare alternative to permanent brain damage that is either paired with another trauma, or risky to get.
If you try to get one through brain damage you also have to keep at least a mild trauma, since by curing one you also cure the special trauma itself.
If you try to do it through lobotomy, you have to be lucky, because you'll have a 99% incurable trauma if you don't roll what you wanted.

**Making lobotomy repeatable and _safe_ until you reroll into a special trauma simply destroys that identity for both lobotomies and special traumas.**

As it is right now, this PR seems made by a powergamer frustrated that they can't simply roll the special traumas reliably and quickly every round.

XRandomXManX commented:
I tried to make this PR a balanced way to limit these special traumas to the Psychologist's Job, giving them a use other than starting with some useless drugs. Then I argue with someone for 4 days, change my PR entirely, and eventually have enough and tell them to fuck off. Then you show up and act like you know what the fuck you're talking about, like your PR overwrites mine. You insult my character and call me disingenuous, even though I was JUST trying to give the Psychologist some more use. The traumas aren't even good! Half the special traumas fucking suck. I didn't want the Psychologist's one role to cause permanent blindness, and instead of saying something about nerfing that, maybe make it so it's nearsightedness, and keep the other severe traumas. I could have done that, the other severe traumas are stuff someone could live with! Why do you have to be so fucking mean to me. It's a code change, it wouldn't do anything to the game. This is so frustrating, I hate this PR. I thought it'd be good but you're all so thoughtless when you say all this shit. I just wanted to do something other than wasting an entire week of my life arguing and having people hate me.

Winterous commented:
I liked this idea from the start, it's a nice little flavour addition; I just disagreed with the 'balance' changes you made. Then you went and made it worse, and worse, and started displaying what struck me as ulterior intent.

Part of the issue here seems to stem from communication difficulty.

Rohesie commented:
> 
> 
> I tried to make this PR a balanced way to limit these special traumas to the Psychologist's Job, giving them a use other than starting with some useless drugs. Then I argue with someone for 4 days, change my PR entirely, and eventually have enough and tell them to fuck off. Then you show up and act like you know what the fuck you're talking about, like your PR overwrites mine. You insult my character and call me disingenuous, even though I was JUST trying to give the Psychologist some more use. The traumas aren't even good! Half the special traumas fucking suck. I didn't want the Psychologist's one role to cause permanent blindness, and instead of saying something about nerfing that, maybe make it so it's nearsightedness, and keep the other severe traumas. I could have done that, the other severe traumas are stuff someone could live with! Why do you have to be so fucking mean to me. It's a code change, it wouldn't do anything to the game. This is so frustrating, I hate this PR. I thought it'd be good but you're all so thoughtless when you say all this shit. I just wanted to do something other than wasting an entire week of my life arguing and having people hate me.

The only maintainer who participated in this discussion is me.
If contributing does not give you some sort of satisfaction, then it's perfectly understandable for you not to do it.
However, do consider I was reviewing your code because I had the intent to merge it.
We can flesh out the balance concerns if you want, feel free to poke me on discord if you don't want to do it here.
If not, thanks for the attempt. Contributors are the lifeblood of this game and the time and effort they spend is invaluable and shouldn't be taken for granted.

Bawhoppen commented:
Do your best to try and also not take any criticism of any changes personally also! People argue all the time about the specific additions and removals to the game, but it's rarely ever meant to be a personal attack against the person doing it. Even though this change didn't go through also, don't give up on contributing either! We all highly value participation in the project, and that's why this codebase has proved to be as successful as it has. I wish you luck on any future changes you might try to make!
